chainstate/src/detail/error.rs

```diff
@@ -60,8 +60,15 @@ pub enum BlockError {
     BlockAtHeightNotFound(BlockHeight),
     #[error("Block {0} already exists")]
     BlockAlreadyExists(Id<Block>),
-    #[error("Failed to commit block state update to database for block: {0} after {1} attempts with error {2}")]
-    DatabaseCommitError(Id<Block>, usize, chainstate_storage::Error),
+    #[error("Block {0} has already been processed")]
+    BlockAlreadyProcessed(Id<Block>),
+    #[error("Block {0} has already been processed and marked as invalid")]
+    InvalidBlockAlreadyProcessed(Id<Block>),
+    #[error("Block {0} has invalid parent block")]
+    InvalidParent(Id<Block>),
+    // Note: the last String here serves as a "comment" that helps distinguish separate commit attempts.
+    #[error("Failed to commit block state update to database for block: {0} after {1} attempts with error {2} ({3})")]
+    DatabaseCommitError(Id<Block>, usize, chainstate_storage::Error, String),
```

Comment:
We don't prefer to have strings in errors because it makes comparing them difficult. It's always better to have a proper enum to represent all possible states, which can be compared with Eq.